src/Integration.Vsix.UnitTests/SonarLintTagger/TextBufferIssueTrackerTests.cs

```diff
@@ -253,231 +261,110 @@ private void CheckAnalysisWasRequested()
+        public void RequestAnalysis_IssueConsumerIsCreatedAndCleared()
         {
-            mockSonarErrorDataSource.Invocations.Clear();
-
-            // Arrange
-            SetupIssuesFilter(out var capturedFilterInput, Enumerable.Empty<IFilterableIssue>());
+            var issueConsumer = new Mock<IIssueConsumer>();
+            var issueConsumerFactory = CreateIssueConsumerFactory(issueConsumer.Object);
 
-            // Act
-            testSubject.HandleNewIssues(Enumerable.Empty<IAnalysisIssueVisualization>());
-
-            // Assert
-            capturedFilterInput.Should().BeEmpty();
-
-            CheckErrorListRefreshWasRequestedOnce(testSubject.Factory);
+            testSubject = CreateTextBufferIssueTracker(issueConsumerFactory: issueConsumerFactory.Object);
 
-            testSubject.Factory.CurrentSnapshot.Issues.Count().Should().Be(0);
-        }
-
-        private void SetupIssuesFilter(out IList<IFilterableIssue> capturedFilterInput,
-            IEnumerable<IFilterableIssue> optionalDataToReturn = null /* if null then the supplied input will be returned */)
-        {
-            var captured = new List<IFilterableIssue>();
-            issuesFilter.Setup(x => x.Filter(It.IsAny<IEnumerable<IFilterableIssue>>()))
-                .Callback((IEnumerable<IFilterableIssue> inputIssues) => captured.AddRange(inputIssues))
-                .Returns(optionalDataToReturn ?? captured);
-            capturedFilterInput = captured;
-        }
+            issueConsumer.Invocations.Clear();
+            issueConsumerFactory.Invocations.Clear();
 
-        private void CheckErrorListRefreshWasRequestedOnce(IssuesSnapshotFactory factory)
-        {
-            mockSonarErrorDataSource.Verify(x => x.RefreshErrorList(factory), Times.Once);
-        }
+            testSubject.RequestAnalysis(Mock.Of<IAnalyzerOptions>());
 
-        private static IAnalysisIssueVisualization CreateIssue(string ruleKey, int startLine, int endLine)
-        {
-            var issue = new DummyAnalysisIssue
-            {
-                RuleKey = ruleKey,
-                PrimaryLocation = new DummyAnalysisIssueLocation
-                {
-                    TextRange = new DummyTextRange
-                    {
-                        StartLine = startLine,
-                        EndLine = endLine,
-                    }
-                }
-            };
-
-            var issueVizMock = new Mock<IAnalysisIssueVisualization>();
-            issueVizMock.Setup(x => x.Issue).Returns(issue);
-            issueVizMock.Setup(x => x.Location).Returns(issue.PrimaryLocation);
-            issueVizMock.Setup(x => x.Flows).Returns(Array.Empty<IAnalysisIssueFlowVisualization>());
-            issueVizMock.SetupProperty(x => x.Span);
-            issueVizMock.Object.Span = new SnapshotSpan(CreateMockTextSnapshot(1000, "any line text").Object, 0, 1);
-
-            return issueVizMock.Object;
-        }
+            // The factory should have been called...
```

Comment:
same comment as above. Maybe it would help to have it in a helper method with a better name, i.e. VerifyNewIssueSnapshotIsCreated and VerifyIssueSnapshotIsCleared and move the comments there